frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/__tests__/k8s-resource.spec.ts

```diff
@@ -59,9 +59,10 @@ describe('k8s-Resource', () => {
   });
 
   it('k8sGetResource should call consoleFetchJSON', async () => {
-    spyCoFetchJSON.mockReturnValueOnce(Promise.resolve({}));
-    await k8sGetResource({ model: MockPodModel, name: 'my-pod' });
+    spyCoFetchJSON.mockReturnValueOnce(Promise.resolve({ kind: 'Pod' }));
+    const result = await k8sGetResource({ model: MockPodModel, name: 'my-pod' });
     expect(spyCoFetchJSON).toHaveBeenCalled();
     expect(spyCoFetchJSON).toHaveBeenCalledTimes(1);
+    expect(result.kind).toBeDefined();
```

Comment:
This was my testing framework to see the type was changed. This is an incredible hack to test through my IDE and not really a valid test from a typing point of view.

We should look into more TypeScript-esk type checking that we can do on the SDK exports. But I'm unsure how different that would be than just desired types against defined types. Maybe a set of use-cases that vet it in a sandbox, more like this? Not sure.

Reply:
Right. I suspect this test would pass before our change as well since we aren't discerning between `Response` or `K8sResourceCommon` right? For now this will still cover us if we regress the K8s func's even worse though :)

Reply:
Right, the test passed before obv. So I'm thinking maybe we need use-case tests for all items exported from the SDK that way we at least have 1 implementation that vets the response to be what we expect it to be... that's really the only way I can think of good tests for this kinda bug.